source/MaterialXRuntime/RtNodeGraph.cpp

```diff
@@ -60,23 +61,30 @@ void RtNodeGraph::removeInput(const RtToken& name)
     prim()->removeAttribute(name);
 }
 
-void RtNodeGraph::renameInput(const RtToken& name, const RtToken& newName)
+RtToken RtNodeGraph::renameInput(const RtToken& name, const RtToken& newName)
 {
     PvtInput* input = prim()->getInput(name);
     if (!(input && input->isA<PvtInput>()))
     {
         throw ExceptionRuntimeError("No input found with name '" + name.str() + "'");
     }
     PvtPrim* socket = prim()->getChild(SOCKETS);
-    socket->renameAttribute(name, newName);
-    prim()->renameAttribute(name, newName);
+    RtToken newPrimAttrName = prim()->renameAttribute(name, newName);
+    RtToken newSocketAttrName = socket->renameAttribute(name, newPrimAttrName);
+    if (newSocketAttrName != newPrimAttrName)
+    {
+        throw ExceptionRuntimeError("Renamed socket name does not match renamed prim name");
```

Comment:
Curious when this would be possible to occur? Should the logic be you rename on the input and then just use that name on the socket similar to the logic in createOutput() ?

Reply:
It should NEVER occur, but it's just a safety check. The logic is the same as createOutput().